client/transaction-pool/src/lib.rs

```diff
@@ -744,3 +913,463 @@ where
+			assert_treeroute_eq(result, expected);
+		}
+
+		#[test]
+		fn tree_route_mock_test_12() {
+			let result = tree_route(d2().hash, b1().hash).expect("tree route exists");
+			let expected = TreeRoute::new(vec![d2(), c2(), b2(), a(), b1()], 3);
+			assert_treeroute_eq(result, expected);
+		}
+
+		#[test]
+		fn tree_route_mock_test_13() {
+			let result = tree_route(c2().hash, e1().hash).expect("tree route exists");
+			let expected = TreeRoute::new(vec![c2(), b2(), a(), b1(), c1(), d1(), e1()], 2);
+			assert_treeroute_eq(result, expected);
+		}
+
+		#[test]
+		fn tree_route_mock_test_14() {
+			let result = tree_route(b1().hash, b1().hash).expect("tree route exists");
+			let expected = TreeRoute::new(vec![b1()], 0);
+			assert_treeroute_eq(result, expected);
+		}
+
+		#[test]
+		fn tree_route_mock_test_15() {
+			let result = tree_route(b2().hash, b2().hash).expect("tree route exists");
+			let expected = TreeRoute::new(vec![b2()], 0);
+			assert_treeroute_eq(result, expected);
+		}
+
+		#[test]
+		fn tree_route_mock_test_16() {
+			let result = tree_route(a().hash, a().hash).expect("tree route exists");
+			let expected = TreeRoute::new(vec![a()], 0);
+			assert_treeroute_eq(result, expected);
+		}
+	}
+
+	use super::EnactmentState;
+	use sc_transaction_pool_api::ChainEvent;
+	use std::sync::Arc;
+
+	fn trigger_new_best_block(
+		state: &mut EnactmentState<Block>,
+		from: HashAndNumber<Block>,
+		acted_on: HashAndNumber<Block>,
+	) -> bool {
+		let (from, acted_on) = (from.hash, acted_on.hash);
+
+		let event_tree_route = tree_route(from, acted_on).expect("Tree route exists");
+
+		state
+			.update_and_check_if_new_enactment_is_valid(
+				&ChainEvent::NewBestBlock {
+					hash: acted_on,
+					tree_route: Some(Arc::new(event_tree_route)),
+				},
+				&tree_route,
+			)
+			.unwrap()
+			.is_some()
+	}
+
+	fn trigger_finalized(
+		state: &mut EnactmentState<Block>,
+		from: HashAndNumber<Block>,
+		acted_on: HashAndNumber<Block>,
+	) -> bool {
+		let (from, acted_on) = (from.hash, acted_on.hash);
+
+		let v = tree_route(from, acted_on)
+			.expect("Tree route exists")
+			.enacted()
+			.iter()
+			.map(|h| h.hash)
+			.collect::<Vec<_>>();
+
+		state
+			.update_and_check_if_new_enactment_is_valid(
+				&ChainEvent::Finalized { hash: acted_on, tree_route: v.into() },
+				&tree_route,
+			)
+			.unwrap()
+			.is_some()
+	}
+
+	fn assert_es_eq(
+		es: &EnactmentState<Block>,
+		expected_best_block: HashAndNumber<Block>,
+		expected_finalized_block: HashAndNumber<Block>,
+	) {
+		assert_eq!(es.recent_best_block, expected_best_block.hash);
+		assert_eq!(es.recent_finalized_block, expected_finalized_block.hash);
+	}
+
+	#[test]
+	fn test_enactment_helper() {
+		sp_tracing::try_init_simple();
+		let mut es = EnactmentState::new(a().hash, a().hash);
+
+		/*
+		 *      B1-C1-D1-E1
+		 *     /
+		 *    A
+		 *     \
+		 *      B2-C2-D2-E2
+		 */
+
+		let result = trigger_new_best_block(&mut es, a(), d1());
+		assert_eq!(result, true);
```

Comment:
```suggestion
		assert!(result);
```
Just a nit to what is considered "idiomatic" by most of the people.

This includes the usage of `//` comments in place of cxx style block comments

Reply:
[32a9522](https://github.com/paritytech/substrate/pull/12305/commits/32a95221007e6c133453706adad91a3e209901c7)